src/main.rs

```diff
@@ -58,7 +49,7 @@ async fn main() {
     let mut settings = match settings {
         Ok(settings) => settings,
         Err(e) => {
-            log::error!("Failed to load settings, continuing with defaults: {:?}", e);
+            tracing::error!("Failed to load settings, continuing with defaults: {:?}", e);
```

Comment:
Failing to find/load external settings and falling back to defaults should probably be a warn over an error depending on the reason for failure. If failure because we failed to parse the settings file, error and probably don't load defaults (or ask if the user would like to continue with defaults) as the user has modified configs and failed to do it properly. If we fail to load settings because file doesn't exist, just warn no file found and continue with defaults. Thoughts?

Reply:
Yes I see your point. I wasn't too keen to have user interaction with the backend application beyond command line args, however I'm not averse to the idea. 

If it errors because it failed to parse that might even be worth an abort to make it absolutely clear to the user that the config is messed up. 

I'm tempted to say that may be out of the scope of this branch though so I might leave it for now. An issue should be made or maybe added as part of the other settings management TODO issue.